scripts/py_matter_idl/examples/matter_idl_plugin/__init__.py

```diff
@@ -0,0 +1,234 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import List
+
+import jinja2
+from matter_idl.generators import CodeGenerator, GeneratorStorage
+from matter_idl.matter_idl_types import Attribute, Cluster, ClusterSide, Command, Field, Idl
+
+
+def toUpperSnakeCase(s):
+    """ Convert camelCaseString to UPPER_SNAKE_CASE with proper handling of acronyms and numerals. """
+    lastLower = False
+    letters = []
+    for c in s:
+        if c.isupper():
+            if lastLower:
+                letters.append('_')
+            letters.append(c)
+            lastLower = False
+        else:
+            letters.append(c.upper())
+            lastLower = True
+
+    return ''.join(letters)
+
+
+def toLowerSnakeCase(s):
+    """ Convert camelCaseString to lower_snake_case with proper handling of acronyms and numerals. """
+    return toUpperSnakeCase(s).lower()
+
+
+def toUpperAcronym(s):
+    """ Remove lower case letters and numbers from the given string"""
+    return ''.join([i for i in s if i.isupper() or i.isnumeric()]).upper()
+
+
+def toEnumEntryName(enumEntry, enumName):
+    """ Create enum entry name by prepending that acronym of the enum name and converting to upper snake case """
+    prefix = toUpperAcronym(enumName)
+    if (enumEntry[0] == 'k'):
+        enumEntry = enumEntry[1:]
+    return prefix + '_' + toUpperSnakeCase(enumEntry)
+
+
+def toProtobufType(zapType: str) -> str:
+    """ Convert zap type to protobuf type """
+    u32Types = ["uint32", "enum8", "enum16", "enum32", "bitmap8", "bitmap16", "bitmap32", "cluster_id", "attrib_id", "event_id", "command_id",
+                "endpoint_no", "group_id", "devtype_id", "fabric_idx", "vendor_id", "status_code", "faulttype", "levelcontroloptions", "percent100ths", "percent"]
+    u64Types = ["uint64", "enum64", "bitmap64", "node_id", "fabric_id", "int40u", "int48u", "int56u", "int64u"]
+    i32Types = ["int32", "int8s", "int16s", "int24s", "int32s"]
+    i64Types = ["int64", "int40s", "int48s", "int56s", "int64s"]
+    floatTypes = ["float", "double"]
+    stringTypes = ["char_string", "long_char_string"]
+    bytesTypes = ["octet_string", "long_octet_string"]
+
+    zapTypeLower = zapType.lower()
+    if zapTypeLower in u32Types:
+        return "uint32"
+    if zapTypeLower in u64Types:
+        return "uint64"
+    if zapTypeLower in i32Types:
+        return "int32"
+    if zapTypeLower in i64Types:
+        return "int64"
+    if zapTypeLower in floatTypes:
+        return "float"
+    if zapTypeLower in stringTypes:
+        return "string"
+    if zapTypeLower in bytesTypes:
+        return "bytes"
+
+    match(zapTypeLower):
```

Comment:
Consider not using match throughout the PR. I believe that forces a newer python version. At least in CI we have:

```
root@ab555ed6ab9d:/workspace# python3 --version
Python 3.9.5
```

(tested on a recent vscode docker image).